pandas/core/tools/datetimes.py

```diff
@@ -446,8 +447,15 @@ def _convert_listlike(arg, box, format, name=None, tz=tz):
             except (ValueError, TypeError):
                 raise e
 
+    def check_numerical_arg():
+        return ((is_scalar(arg) and (is_integer(arg) or is_float(arg))) or
```

Comment:
Without it, `pd.to_datetime([])` raises a ValueError("a unit is required in case of numerical arg") which is not the actual behaviour in some others tests (`pandas/tests/test_resample.py::TestPeriodIndex::test_resample_empty_dtypes` for example).